src/orders/components/OrderPaymentOrTransaction/OrderTransactionsWrapper.tsx

```diff
@@ -0,0 +1,90 @@
+import CardSpacer from "@dashboard/components/CardSpacer";
+import { TransactionActionEnum } from "@dashboard/graphql/types.generated";
+import {
+  OrderDetailsWithTransactionsFragment,
+  OrderDetailsWithTransactionsQuery,
+} from "@dashboard/graphql/types.transactions.generated";
+import React from "react";
+
+import OrderAddTransaction from "../OrderAddTransaction";
+import { useStyles } from "../OrderDetailsPage/styles";
+import OrderGrantedRefunds from "../OrderGrantedRefunds";
+import OrderPaymentSummaryCard from "../OrderPaymentSummaryCard";
+import OrderSummaryCard from "../OrderSummaryCard";
+import OrderTransaction from "../OrderTransaction";
+import OrderTransactionGiftCard from "../OrderTransactionGiftCard";
+import OrderTransactionPayment from "../OrderTransactionPayment";
+
+interface OrderTransactionsWrapper {
+  order: OrderDetailsWithTransactionsFragment;
+  shop: OrderDetailsWithTransactionsQuery["shop"];
+  onTransactionAction(transactionId: string, actionType: TransactionActionEnum);
+  onPaymentCapture();
+  onPaymentPaid();
+  onPaymentVoid();
+  onAddManualTransaction();
+}
+
+export const OrderTransactionsWrapper: React.FC<OrderTransactionsWrapper> = ({
+  order,
+  shop,
+  onTransactionAction,
+  onPaymentCapture,
+  onPaymentPaid,
+  onPaymentVoid,
+  onAddManualTransaction,
+}) => {
+  const classes = useStyles();
+
+  const filteredPayments = React.useMemo(
+    () =>
+      (order?.payments ?? []).filter(
+        payment => payment.isActive || payment.transactions.length > 0,
+      ),
+    [order?.payments],
+  );
```

Comment:
**suggestion:** It's unclear what this does, condition together with nullish value handling makes it difficult to read. Please try using more explicit naming, perhaps it could be extracted and/or unit tested.